imagetagger/imagetagger/images/views.py

```diff
@@ -638,6 +639,48 @@ def dl_script(request):
                             }, content_type='text/plain')
 
 
+def download_imageset_zip(request, image_set_id):
+    image_set = get_object_or_404(ImageSet, id=image_set_id)
+
+    if not image_set.has_perm('read', request.user):
+        return HttpResponseForbidden()
+
+    if image_set.image_count == 0:
+        # It should not be possible to download empty image sets. This
+        # is already blocked in the UI, but it should also be checked
+        # on the server side.
+        messages.warning(request, 'There are no images in the selected image set.')
+        return redirect(reverse('images:view_imageset', args=(image_set_id,)))
+
+    if not image_set.zip_finished:
+        # The zip creation process after an upload is still in progress.
+        # Therefore, we do not want to start a second process operating
+        # on the same file.
+        messages.warning(request, 'The zip file generation is still in progress. Try again in a few minutes.')
+        return redirect(reverse('images:view_imageset', args=(image_set_id,)))
```

Comment:
I think this should return a http error code to make the download via scripts possible